src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -543,6 +543,110 @@ public void UserDefinedFunctionTest(string udfExpression, string expression, boo
+            "distance({x: 0, y: 0}, {x: 0, y: 5})",
+            true,
+            5.0)]
+
+        // Table types are accepted
+        [InlineData(
+            "People = Type([{Id:Number, Age: Number}]); countMinors(p: People): Number = CountRows(Filter(p, Age < 18));",
+            "countMinors([{Id: 1, Age: 17}, {Id: 2, Age: 21}])",
+            true,
+            1.0)]
+        [InlineData(
+            "Numbers = Type([Number]); countEven(nums: Numbers): Number = CountRows(Filter(nums, Mod(Value, 2) = 0));",
+            "countEven([1,2,3,4,5,6,7,8,9,10])",
+            true,
+            5.0)]
+
+        // Type Aliases are allowed
+        [InlineData(
+            "CarYear = Type(Number); Car = Type({Model: Text, ModelYear: CarYear}); createCar(model:Number, year: Number): Car = {Model:model, ModelYear: year};",
+            "createCar(\"Model Y\", 2024).ModelYear",
+            true,
+            2024.0)]
+
+        // Type definitions order shouldn't matter
+        [InlineData(
+            "Person = Type({Id: IdType, Age: Number}); IdType = Type(Number); createUser(id:Number, a: Number): Person = {Id:id, Age: a};",
+            "createUser(1, 42).Age",
+            true,
+            42.0)]
+
+        // Functions accept record with more/less fields
+        [InlineData(
+            "People = Type([{Name: Text, Age: Number}]); countMinors(p: People): Number = CountRows(Filter(p, Age < 18));",
+            "countMinors([{Name: \"Bob\", Age: 21, Title: \"Engineer\"}, {Name: \"Alice\", Age: 25, Title: \"Manager\"}])",
+            true,
+            0.0)]
+        [InlineData(
+            "Employee = Type({Name: Text, Age: Number, Title: Text}); getAge(e: Employee): Number = e.Age;",
+            "getAge({Name: \"Bob\", Age: 21})",
+            true,
+            21.0)]
+        [InlineData(
+            @"Employee = Type({Name: Text, Age: Number, Title: Text}); Employees = Type([Employee]);  EmployeeNames = Type([{Name: Text}]); 
+              getNames(e: Employees):EmployeeNames = ShowColumns(e, Name); 
+              getNamesCount(e: EmployeeNames):Number = CountRows(getNames(e));",
+            "getNamesCount([{Name: \"Jim\", Age:25}, {Name: \"Tony\", Age:42}])",
+            true,
+            2.0)]
+        [InlineData(
+            @"Employee = Type({Name: Text, Age: Number, Title: Text}); 
+              getAge(e: Employee): Number = e.Age;
+              hasNoAge(e: Employee): Number = IsBlank(getAge(e));",
+            "hasNoAge({Name: \"Bob\", Title: \"CEO\"})",
+            true,
+            1.0)]
+
+        // Cycles not allowed
+        [InlineData(
+            "Z = Type([{a: {b: Z}}]);",
+            "",
+            false)]
+        [InlineData(
+            "X = Type(Y); Y = Type(X);",
+            "",
+            false)]
+        [InlineData(
+            "C = Type({x: Boolean, y: Date, f: B});B = Type({ x: A }); A = Type([C]);",

```

Comment:
>[C] [](http://example.com/codeflow?start=80&length=3)

Is this array of types?

Reply:
it is Table of type C, using `[ ]` syntax for table types